extensions/amp-share-tracking/0.1/amp-share-tracking.js

```diff
@@ -15,43 +15,116 @@
  */
 
 import {isExperimentOn} from '../../../src/experiments';
-import {getService} from '../../../src/service';
-import {dev} from '../../../src/log';
+import {xhrFor} from '../../../src/xhr';
+import {viewerFor} from '../../../src/viewer';
+import {dev, user} from '../../../src/log';
 
 /** @private @const {string} */
 const TAG = 'amp-share-tracking';
 
-class AmpShareTracking extends AMP.BaseElement {
+export class AmpShareTracking extends AMP.BaseElement {
+  /**
+    * @return {boolean}
+    * @private
+    */
+  isExperimentOn_() {
+    return isExperimentOn(this.win, TAG);
+  }
+
+  /** @override */
+  isLayoutSupported(unusedLayout) {
+    return true;
+  }
 
   /** @override */
   buildCallback() {
-    this.isExperimentOn_ = isExperimentOn(this.win, TAG);
-    if (!this.isExperimentOn_) {
+    if (!this.isExperimentOn_()) {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+
+    /** @private {string} */
+    this.vendorHref_ = this.element.getAttribute('data-href');
+    dev.fine(TAG, 'vendorHref_: ', this.vendorHref_);
+
+    /** @private {!Promise<!Object>} */
+    this.shareTrackingFragments_ = Promise.all([
+      this.getIncomingFragment_(),
+      this.getOutgoingFragment_(this.vendorHref_)]).then(results => {
```

Comment:
Why pass an instance variable to a instance method? `getOutgoingFragment_` has the access to `this.vendorHref_` by itself.